UI/Components/TheoryComparisonGeneratorSettings.cs

```diff
@@ -345,33 +345,43 @@ private void _unlockFile()
                 File.Delete(TheoryTimesFilePathLock);
             }
         }
-
-        private void _theoryComparisonGeneratorSettings_Leave(object sender, EventArgs e)
+        private void _LayoutSettingsClosed(object sender, EventArgs e)
         {
             var shouldSave = false;
 
-            if (HasChanged)
-            {
-                var result = MessageBox.Show(
-                    "Would you like to save changes made to theory times file?",
-                    "Save Changes",
-                    MessageBoxButtons.YesNo,
-                    MessageBoxIcon.Question
-                );
-
-                shouldSave = result == DialogResult.Yes;
-            }
+                if (HasChanged)
+                {
+                    var result = MessageBox.Show(
+                        "Would you like to save changes made to the theory times file?",
+                        "Save Changes",
+                        MessageBoxButtons.YesNo,
+                        MessageBoxIcon.Question
+                    );
 
-            if (fileLock != null)
-            {
-                if (shouldSave)
+                    shouldSave = result == DialogResult.Yes;
+                }
+
+                if (fileLock != null)
+                {
+                    if (shouldSave)
+                        _writeTheoryTimesFilePath();
+                    else
+                        _loadTheoryTimesFilePath(TheoryTimesFilePath);
+                }
+                else if (shouldSave && !string.IsNullOrEmpty(TheoryTimesFilePath) && !File.Exists(TheoryTimesFilePath))
```

Comment:
idea: hoping this branch is due to double unlock, i dont think it should be necessary if lock is not early removed. In any case i think we should never reach a shouldSave state where we dont have a lock on the file because that would mean we made a change to the file contents without having the right to. Does that make sense?